File: src/main/java/io/vizit/vpoc/jvm/model/JvmConfig.java
package io.vizit.vpoc.jvm.model;

public class JvmConfig {
    public static final int Xms = 300; // minimum heap size
    public static final int Xmx = Xms; // maximum heap size
    public static final int SurvivorRatio = 8; // survivor:eden = 1:8
    public static final int NewRatio = 2;// young:old = 1:2

    public static final int getYoungSize() {
        return Xmx * 1 / (JvmConfig.NewRatio + 1);
    }

    public static final int getOldSize() {
        return Xmx - getYoungSize();
    }

    public static final int getEdenSize() {
        return getYoungSize() * JvmConfig.SurvivorRatio / (JvmConfig.SurvivorRatio + 2);
    }

    public static final int getSurvivorSize() {
        return (getYoungSize() - getEdenSize()) / 2;
    }
}

File: src/main/resources/static/scripts/jvm_config.js
export const gc_config = {
    SurvivorRatio: 8, // survivor:eden = 1:8
    NewRatio: 2 // young:old = 1:2
}
export const obj_space = {
    width: 80,
    height: 40
}
export const eden_space = {
    width: 800,
    height: 320,
}
export const survivor_space = {
    width: eden_space.width / 1,
    height: eden_space.height / 8
}
export const old_space = {
    width: eden_space.width,
    height: eden_space.height * gc_config.NewRatio
}